proto/private/protoc/mirror_protoc_release.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+# By default, mirrors the most recent release of protoc from
+# https://github.com/protocolbuffers/protobuf/releases
+#
+# To mirror a different version, set VERSION in the environment.
+# To use a different fork, set REPOSITORY in the environment.
+set -o nounset -o errexit -o pipefail
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+REPOSITORY=${REPOSITORY:-"protocolbuffers/protobuf"}
+VERSIONS_BZL="$SCRIPT_DIR/versions.bzl"
+
+RELEASES=$(mktemp)
+curl > $RELEASES \
+  --silent \
+  --header "Accept: application/vnd.github.v3+json" \
+  https://api.github.com/repos/${REPOSITORY}/releases?per_page=50
+
+# If $VERSION unset, mirror the latest release
+VERSION=${VERSION:-$(jq --raw-output '.[0].tag_name' $RELEASES)}
+echo "Mirroring integrity hashes for $REPOSITORY at release $VERSION ..."
+
+DOWNLOAD_URLS_FILTER='
+map(select(.tag_name == $version))[0]
+| .assets
+| map(select(.name | startswith("protoc")))
+| map(.browser_download_url)[]
+'
+# Workaround: protobuf doesn't publish their integrity hashes to e.g. checksums.txt
+# Create a file that looks like a checksums.txt from a shasum command, i.e.
+# sha384-RVFu8PJJCOSXwYTqH7FyWRSgsP1AAjcEa+VViddVTgtd9wYvZjQoQ8jmlFxwfFw+ protobuf-26.0-rc3.tar.gz
+# sha384-JYSXGTSBfwUU6UzqazUTkT3lTZDzx10YdaNQYjojrT7X1Ro1fA+T4tjJw0e8UISV protobuf-26.0-rc3.zip
+CHECKSUMS=$(mktemp)
+for url in $(jq --arg version $VERSION --raw-output "$DOWNLOAD_URLS_FILTER" <$RELEASES); do
+    sha=$(curl -sSL $url | shasum -b -a 384 | awk "{ print \$1 }" | xxd -r -p | base64)
+    echo "sha384-${sha} $(basename $url)"
+done | tee "$CHECKSUMS"
```

Comment:
Downloading an computing hashes doesn't feel very secure / the right thing to do.

Reply:
It's https://en.wikipedia.org/wiki/Trust_on_first_use - we assume that within 24 hours of a protobuf release, the files they are serving are still uncompromised.

I can leave a TODO here to use the checksums.txt file if the protobuf project starts publishing one. WDYT?